src/Disks/ObjectStorages/IObjectStorage.cpp

```diff
@@ -47,4 +47,27 @@ std::string IObjectStorage::getCacheBasePath() const
     throw Exception(ErrorCodes::NOT_IMPLEMENTED, "getCacheBasePath() is not implemented for {}", getName());
 }
 
+void IObjectStorage::applyRemoteThrottlingSettings(ContextPtr context)
+{
+    std::unique_lock lock{throttlers_mutex};
+    read_throttler = context->getRemoteReadThrottler();
+    write_throttler = context->getRemoteWriteThrottler();
```

Comment:
btw not there is also `LocalObjectStorage`, so using `getRemote*` as base implementation might be confusing. But I think for now it is ok, in the future we can just add a check here `if (isRemote()) ... else ...`.

Reply:
ah, I see this method is called `applyRemote..`, so do not mind this comment.